addrmgr/addrmanager.go

```diff
@@ -483,20 +483,20 @@ func (a *AddrManager) savePeers() {
 	tmpfile := a.peersFile + ".new"
 	w, err := os.Create(tmpfile)
 	if err != nil {
-		log.Errorf("Error opening file %s: %v", tmpfile, err)
+		log.Errorf("Error opening file %s: %w", tmpfile, err)
 		return
```

Comment:
Uhmm, I don't think we need to wrap logged errors here and below.

Reply:
Good catch.  Looks like I got a bit overzealous on the find/replace there.  Reverted this and the other cases.

Reply:
Sorry to comment on an old commit, but I'm just trying to understand better "idiomatic" go error handling. Why *not* wrap every possible error?

Reply:
Technically speaking, logging writes the error to a raw output stream and thus there is no way for the other side to unwrap it since all type information is lost in such an operation.

However, philosophically speaking, the entire point of wrapping errors is so that something further up the call stack can identify the underlying error programmatically.  You don't wrap errors when you pass them _down_ to something else.

Finally, practically speaking, not all logging implementations even understand the `%w` formatting verb because it wasn't added until Go 1.13 (I think, it's been a while) and there is something like 7+ years of code written prior to that.